src/pytorch_lightning/core/optimizer.py

```diff
@@ -176,21 +176,22 @@ def _init_optimizers_and_lr_schedulers(
         )
         optim_conf = _MockOptimizer()
 
-    optimizers, lr_schedulers, optimizer_frequencies, monitor = _configure_optimizers(optim_conf)
+    optimizers, lr_schedulers, monitor = _configure_optimizers(optim_conf)
     lr_scheduler_configs = (
         _configure_schedulers_automatic_opt(lr_schedulers, monitor)
         if model.automatic_optimization
         else _configure_schedulers_manual_opt(lr_schedulers)
     )
-    _set_scheduler_opt_idx(optimizers, lr_scheduler_configs)
+    _validate_multiple_optimizers_support(optimizers, model)
+    _validate_optimizers_attached(optimizers, lr_scheduler_configs)
     _validate_scheduler_api(lr_scheduler_configs, model)
-    return optimizers, lr_scheduler_configs, optimizer_frequencies
+    return optimizers, lr_scheduler_configs
```

Comment:
couldn't you still have a single optimizer frequency which would then be equivalent to gradient accumulation?

Reply:
If we did this, there would be two ways of enabling gradient accumulation. Isn't the Trainer argument more intuitive? It's an interesting idea though.

Reply:
I think the trainer argument is more intuitive, however, for me this is an inherent property of the module...

Reply:
I thought about this and it actually doesn't work with only a single optimizer. Since we execute the training_step as part of the closure, Not doing the step would mean that the entire training step doesn't get executed. 

Frequency made sense when we had multiple, because the training_step closure would always be executed by either one of the optimizers, and because the user had guarded their code with the optimizer_idx, this would work out.